src/application/common/endpoint.ts

```diff
@@ -24,7 +27,21 @@ export class Endpoint {
     }
 
     protected get host() {
-        return location.host || "127.0.0.1:3000"
+
+
+        if (location.host === "") {
+            this.fs.readFile('port', "utf8", function (err: string, data: any) {
```

Comment:
for electron I meant something like:
```ts
require("../backend/main").then(app => {
    mainWindow = new electron.BrowserWindow({ width: 1024, height: 728 });
    mainWindow.webContents.openDevTools();
    mainWindow.loadURL(`http://localhost:${app.port}`);
    // or as a parameter to a file URI
    // mainWindow.loadURL(`file://${path.join(__dirname, '../../lib/index.html')}?port=${app.port}`);
    mainWindow.on('closed', function () {
        mainWindow = undefined;
    });
});
```

where `app.port` is implemented via `server.address().port`